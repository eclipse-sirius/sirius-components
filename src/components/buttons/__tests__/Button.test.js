/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import Renderer from 'react-test-renderer';

import {
  BUTTON_DANGER__KIND,
  BUTTON_DEFAULT__KIND,
  BUTTON_PRIMARY__KIND,
  BUTTON_SUCCESS__KIND,
  BUTTON_WARNING__KIND,
  Button
} from '../Button';

describe('Button', () => {
  it('renders a button without a label', () => {
    const button = Renderer.create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a button with a label', () => {
    const button = Renderer.create(<Button>OK</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a default button', () => {
    const button = Renderer.create(<Button kind={BUTTON_DEFAULT__KIND} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a primary button', () => {
    const button = Renderer.create(<Button kind={BUTTON_PRIMARY__KIND} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a success button', () => {
    const button = Renderer.create(<Button kind={BUTTON_SUCCESS__KIND} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a warning button', () => {
    const button = Renderer.create(<Button kind={BUTTON_WARNING__KIND} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('renders a danger button', () => {
    const button = Renderer.create(<Button kind={BUTTON_DANGER__KIND} />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  describe('Handles DOM events', () => {
    it('can trigger the onClick event handler', () => {
      let hasBeenClicked = false;

      const div = document.createElement('div');
      ReactDOM.render(<Button onClick={() => (hasBeenClicked = true)}>OK</Button>, div);

      const buttonDOMNode = div.childNodes[0];

      ReactTestUtils.Simulate.click(buttonDOMNode);
      expect(hasBeenClicked).toEqual(true);
    });
  });
});
