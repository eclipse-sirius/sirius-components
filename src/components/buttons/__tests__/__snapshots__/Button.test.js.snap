// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button with a label 1`] = `
<button
  className="button button--default"
>
  OK
</button>
`;

exports[`Button renders a button without a label 1`] = `
<button
  className="button button--default"
/>
`;

exports[`Button renders a danger button 1`] = `
<button
  className="button button--danger"
/>
`;

exports[`Button renders a default button 1`] = `
<button
  className="button button--default"
/>
`;

exports[`Button renders a primary button 1`] = `
<button
  className="button button--primary"
/>
`;

exports[`Button renders a success button 1`] = `
<button
  className="button button--success"
/>
`;

exports[`Button renders a warning button 1`] = `
<button
  className="button button--warning"
/>
`;
